### PR TITLE
Improving network status detection

### PR DESCRIPTION
**Pull request by antonmolodykh**
Refs https://github.com/Screenly/screenly-ose/issues/579

### CODE REVIEW

File: lib/utils.py [hunk 1]
```diff
@@ -56,18 +56,33 @@ def validate_url(string):
...
-        my_ip = ifaddresses(default_interface)[2][0]['addr']
-        return my_ip
-    except:
-        pass
+        if 'eth0' in active_interfaces:
+            interface = 'eth0'
+        elif 'wlan0' in active_interfaces:
+            interface = 'wlan0'
+        else:
+            raise Exception("Display error saying no active network interface found")
```
Comment (vpetersson) [inline lib/utils.py]: 'No active network interface found.'

File: lib/utils.py [hunk 2]
```diff
@@ -56,18 +56,33 @@ def validate_url(string):
...
+
+        try:
+            my_ip = ifaddresses(interface)[2][0]['addr']
+            return my_ip
+        except KeyError:
+            raise Exception("Display error saying that we were unable to retrieve an IP")
```
Comment (vpetersson) [inline lib/utils.py]: 'Unable to retrieve an IP.'

File: lib/utils.py [hunk 3]
```diff
@@ -56,18 +56,33 @@ def validate_url(string):
...
+        """Returns the node's IP, for the interface
+        that is being used as the default gateway.
+        This shuld work on both MacOS X and Linux."""
+
+        try:
+            default_interface = grep(netstat('-nr'), '-e', '^default', '-e' '^0.0.0.0').split()[-1]
+            my_ip = ifaddresses(default_interface)[2][0]['addr']
+            return my_ip
+        except:
+            raise Exception("Unable to look up your installation's IP address.")
```
Comment (vpetersson) [inline lib/utils.py]: 'Unable to resolve local IP address.'

File: server.py [hunk 4]
```diff
@@ -382,8 +382,12 @@ def system_info():
 
 @route('/splash_page')
 def splash_page():
-    my_ip = get_node_ip()
-    if my_ip:
+    try:
+        my_ip = get_node_ip()
+    except Exception as e:
+        ip_lookup = False
+        url = e
```
Comment (vpetersson) [inline server.py]: `url` seems to be an odd choice for this variable. How about `error_msg` or something like that instead?

File: lib/utils.py (hunk 1 above)
Comment (antonmolodykh) [inline lib/utils.py]: Oh, my mistake. I was in a hurry and just copied

**Pull request merged**